SetFlags.cmake

```diff
@@ -62,6 +62,9 @@ macro(set_flags)
 		set(CMAKE_EXE_LINKER_FLAGS_RELEASE    "${CMAKE_EXE_LINKER_FLAGS_RELEASE}    /LTCG")
 		set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${CMAKE_SHARED_LINKER_FLAGS_RELEASE} /LTCG")
 		set(CMAKE_MODULE_LINKER_FLAGS_RELEASE "${CMAKE_MODULE_LINKER_FLAGS_RELEASE} /LTCG")
+
+		# Make build use Unicode:
+		add_definitions(-DUNICODE -D_UNICODE)
```

Comment:
By we use utf-8, I'm referring to what our strings are encoded with. So all of our code expects a std::string to be in an encoding that uses single byte encoding units. In practice this means utf-8 or a windows code-page, but for a lot of string manipulation it doesn't actually matter which. 

UCS-2/UTF-16 is different enough that it can cause serious issues if it is passed to a function that expects a single byte encoding unit encoding. For example UCS-2/UTF-16 can have single byte embedded zero bytes in the middle of text, which breaks strcpy.